### DISCUSSION TITLE
RESTful Deeplinking?

### DISCUSSION THREAD

**Discussion by assassin316** [unoplatform/uno#14111] [Q&A] [answered]
In regards to deep-linking. Uno easily supports the following type of deep-link using the **Application.OnLaunched** event:

ex:
www.mysite.com/?page1

This is based on a discussion here:
https://stackoverflow.com/a/61604579

If the ? is omitted, the page1 value doesn't come through in the arguments. Is there a way to capture restful deep-linking?

ex:
www.website.com/page1

In other words, do we have support for RESTful deeplinking?

**Comment by jeromelaban** [comment, marked as answer]
At this time, we have not found a good way to include this in the standard WinUI APIs. That being said, you should be able to use something like:

```csharp
 private static partial class NativeMethods
 {
     [JSImport("globalThis.getLocation()")]
     public static partial string GetCurrentLocation();
 }
```

Then have a function in javascript with this:
```js
function getLocation() {
  return window.location.href;
}
```

Then route explicitly with this. Note that Uno.Extensions handles this part automatically, if you use the navigation extension.

**Comment by assassin316** [comment]
Perfect, that will do, thank you 👍
